src/main/java/seedu/clinic/logic/parser/EditCommandParser.java

```diff
@@ -45,51 +46,56 @@
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_SUPPLIER_INDEX, PREFIX_WAREHOUSE_INDEX,
-                        PREFIX_SUPPLIER_NAME, PREFIX_WAREHOUSE_NAME,
+                ArgumentTokenizer.tokenize(args, PREFIX_TYPE, PREFIX_INDEX, PREFIX_NAME,
                         PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_REMARK);
 
         logger.log(Level.INFO, "Successfully tokenized user input.");
 
-        if (argMultimap.getValue(PREFIX_SUPPLIER_INDEX).isPresent()
-                && argMultimap.getValue(PREFIX_WAREHOUSE_INDEX).isPresent()) {
-            throw new ParseException(String.format(MESSAGE_INPUT_BOTH_SUPPLIER_WAREHOUSE_PREFIX,
+        if (!argMultimap.getValue(PREFIX_TYPE).isPresent()
+                && !argMultimap.getValue(PREFIX_INDEX).isPresent()) {
```

Comment:
I think the original way is to define a new function called arePrefixsPresent (maybe you can check the deleteCommandParser and addCommandParser). Not sure if this is the new form intended

Reply:
Just went to take a look at how it is defined in deleteCommandParser and addCommandParser, and yes I think arePrefixsPresent does these checks! I can change it to this function, However, I realised that would also mean I will not be able to throw specific ParseException reminding user which prefixes they are missing (either one, or all). Which do you think would be better (in terms of code quality and user experience?) :)